src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/sessiondays/SessionDaysReport.java

```diff
@@ -0,0 +1,271 @@
+        List<SessionDaysReportDetail> sessionDaysReportDetailList = new ArrayList<>();
+        for (SessionDaysSubmitEvent submitEvent : submitEvents) {
+            var caseData = submitEvent.getCaseData();
+            setCaseReportSummaries(caseData, reportData.getReportSummary(), sessionDaysReportSummary2List);
+            setReportDetail(caseData, sessionDaysReportDetailList);
+        }
+        int ft = Integer.parseInt(reportData.getReportSummary().getFtSessionDaysTotal());
+        int pt = Integer.parseInt(reportData.getReportSummary().getPtSessionDaysTotal());
+        int ot = Integer.parseInt(reportData.getReportSummary().getOtherSessionDaysTotal());
+        int total = ft + pt + ot;
+        int ptPercent = total > 0 ? (pt * 100) / total : 0;
+        reportData.getReportSummary().setSessionDaysTotal(String.valueOf(total));
+        reportData.getReportSummary().setPtSessionDaysPerCent(String.valueOf(ptPercent));
+        reportData.addReportSummary2List(sessionDaysReportSummary2List);
+        reportData.addReportDetail(sessionDaysReportDetailList);
+    }
+
+    private SessionDaysReportSummary2 getReportSummary2Item(
+            DateListedType dateListedType, List<SessionDaysReportSummary2> sessionDaysReportSummary2List) {
+        Optional<SessionDaysReportSummary2> item = sessionDaysReportSummary2List.stream()
+                .filter(i -> !Strings.isNullOrEmpty(i.getDate())
+                && i.getDate().equals(dateListedType.getListedDate())).findFirst();
+        if (item.isPresent()) {
+            return item.get();
+        }
+        SessionDaysReportSummary2 summary2 = new SessionDaysReportSummary2();
+        initReportSummary2(summary2);
+        summary2.setDate(dateListedType.getListedDate());
+        sessionDaysReportSummary2List.add(summary2);
+        return summary2;
+    }
+
+    private void setCaseReportSummaries(SessionDaysCaseData caseData,
+                                        SessionDaysReportSummary reportSummary,
+                                        List<SessionDaysReportSummary2> sessionDaysReportSummary2List) {
+        int ft;
+        int pt;
+        int ot;
+        int ft2 = 0;
+        int pt2 = 0;
+        int ot2 = 0;
+        for (HearingTypeItem hearingTypeItem : getHearings(caseData)) {
+            var dates = hearingTypeItem.getValue().getHearingDateCollection();
+            if (CollectionUtils.isNotEmpty(dates)) {
+                for (DateListedTypeItem dateListedTypeItem : dates) {
+                    if (isHearingStatusValid(dateListedTypeItem)) {
+                        String judgeStatus = getJudgeStatus(hearingTypeItem.getValue().getJudge());
+                        SessionDaysReportSummary2 reportSummary2 = getReportSummary2Item(
+                                dateListedTypeItem.getValue(), sessionDaysReportSummary2List);
+                        switch (judgeStatus) {
+                            case "SALARIED":
```

Comment:
Having created an enum for judge employment status then you really ought to use it

Reply:
using enum now